changelogs/unreleased/6704-27149chen

```diff
@@ -0,0 +1,3 @@
+This pr made some improvements in Resource Modifiers:
+1. add label selector
+2. change the field name from groupKind to groupResource
```

Comment:
can you please explain your rationale for the change to groupResource?
I preferred groupKind for clarity in K8s terms.
is there any precedence for groupResource?

Reply:
got it thanks for clarifying.